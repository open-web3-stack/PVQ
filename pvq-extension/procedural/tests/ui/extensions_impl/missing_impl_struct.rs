use pvq_extension_procedural::{extension_decl, extensions_impl};

#[extension_decl]
pub mod test_extension {
    #[extension_decl::extension]
    pub trait TestExtension {
        type Value;
        fn test_fn(value: Self::Value) -> bool;
    }
}

#[extensions_impl]
mod test_impl {
    // Missing #[extensions_impl::impl_struct] attribute
    pub struct TestImpl;

    #[extensions_impl::extension]
    impl test_extension::TestExtension for TestImpl {
        type Value = u32;
        fn test_fn(value: u32) -> bool {
            value > 0
        }
    }
}

fn main() {}
