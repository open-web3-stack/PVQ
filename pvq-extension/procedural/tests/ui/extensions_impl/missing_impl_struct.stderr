error: Missing `#[extensions_impl::impl_struct]`
  --> tests/ui/extensions_impl/missing_impl_struct.rs:13:1
   |
13 | mod test_impl {
   | ^^^
